mantis-runtime/src/main/java/io/mantisrx/runtime/core/MantisStream.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.runtime.core;
+
+import io.mantisrx.runtime.core.functions.FilterFunction;
+import io.mantisrx.runtime.core.functions.FlatMapFunction;
+import io.mantisrx.runtime.core.functions.KeyByFunction;
+import io.mantisrx.runtime.core.functions.MapFunction;
+import io.mantisrx.runtime.core.functions.ReduceFunction;
+import io.mantisrx.runtime.core.sinks.SinkFunction;
+import io.mantisrx.runtime.core.sources.SourceFunction;
+import io.mantisrx.runtime.parameter.ParameterDefinition;
+import java.time.Duration;
+import lombok.Getter;
+
+public interface MantisStream<T> {
+
+    <OUT> MantisStream<OUT> source(SourceFunction<OUT> sourceFunction);
+
+    MantisStream<Void> sink(SinkFunction<T> sinkFunction);
```

Comment:
What's the reason that this is producing an output? Shouldn't this be a void function?

For instance, I'm guessing you want users to avoid doing things like
```
MantisStream.source().sink().sink() etc...?
```